src/hba.c

```diff
@@ -499,7 +467,143 @@ static bool bad_mask(struct HBARule *rule)
 	return !!res;
 }
 
-static bool parse_line(struct HBA *hba, struct TokParser *tp, int linenr, const char *parent_filename)
+static bool match_map(struct HBARule *rule, struct IDENT *ident, const char *mapname)
+{
+	struct List *el;
+	struct IDENTMap *map;
+
+	if (!ident)
+		return false;
+
+	list_for_each(el, &ident->maps) {
+		map = container_of(el, struct IDENTMap, node);
+
+		if (strcmp(map->map_name, mapname) == 0) {
+			rule->identmap = map;
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool parse_map_definition(struct HBARule *rule, struct IDENT *ident, struct TokParser *tp, int linenr)
+{
+	const char *mapdef;
+	const char *mapname = NULL;
+	const char *rest;
+
+	if (!expect(tp, TOK_IDENT, &mapdef))
+		return true;
+
+	log_warning("hba line %d: map definition : %s", linenr, mapdef);
+
+	if (strcmp(mapdef, "map") == 0) {
+		next_token(tp);
+
+		if (!expect(tp, TOK_IDENT, &rest))
+			return false;
+
+		if (strcmp(rest, "=") == 0) {
+			next_token(tp);
+
+			if (!expect(tp, TOK_IDENT, &mapname))
+				return false;
+		} else if (rest[0] == '=') {
+			mapname = rest + 1;
+		} else {
+			return false;
+		}
```

Comment:
Afaict this logic is to handle spaces between the `=` sign an `map`. This is unnecessary, postgres also does not support this: https://github.com/postgres/postgres/blob/master/src/backend/libpq/hba.c#L1856-L1884

Also, as you can see in the `parse_hba_auth_opt` function, Postgres actually supports a bunch of these options in hba lines. Let's at least prepare our code to do the same (i.e. using `strchr` to get the option name). That way we can give more useful error messages, e.g. option clientcert is not supported.